packages/jsii-runtime/lib/host.ts

```diff
@@ -14,7 +14,7 @@ export class KernelHost {
             return; // done
         }
 
-        this.processRequest(req, () => this.run());
+        this.processRequest(req, () => setImmediate(() => this.run()));
```

Comment:
I wonder if this should be inside processRequest
Also add a comment explaining why this is needed

Reply:
I wasn't sure whether it belongs inside run() or processRequest(). I went with this because it seemed to be less invasive:

The result of the next() call is returned from processRequest(), and this wouldn't be possible with a setImmediate() call.  This would affect callbackHandler()/completeCallback() because it returns processRequest()'s result. Limiting the change to run() doesn't affect anything because run() discards processRequest()'s result.

Is that too cautious?

If the behavior change is acceptable, then moving it to processRequest() would be nice.

Reply:
Got it. Let’s leave it like this. Just add some comments to explain why this is needed and we are golden.  Thanks for the quality contribution!

Reply:
Thank you! I've added a comment.